packages/resolver-extensions/src/uri_resolver_wrapper.rs

```diff
@@ -89,25 +89,114 @@ impl UriResolver for UriResolverWrapper {
             invoker
         );
 
+        let uri = if let Some(resolved_uri) = result.uri {
+            Uri::try_from(resolved_uri)?
+        } else {
+            uri.clone()
+        };
+
         if let Some(manifest) = result.manifest {
             let package = WasmPackage::new(
                 Arc::new(file_reader),
                 Some(manifest),
                 None
             );
-            return Ok(UriPackageOrWrapper::Package(uri.clone(), Arc::new(package)));
-        }
-
-        if let Some(uri) = result.uri {
-            return Ok(UriPackageOrWrapper::Uri(uri.try_into()?));
+            return Ok(UriPackageOrWrapper::Package(
+                uri.clone(), 
+                Arc::new(package
+            )));
         }
 
-        Ok(UriPackageOrWrapper::Uri(uri.clone()))
+        Ok(UriPackageOrWrapper::Uri(uri))
     }
 }
 
 impl fmt::Debug for UriResolverWrapper {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "UriResolverWrapper: {:?}", self.implementation_uri)
     }
-}
\ No newline at end of file
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::{Arc, Mutex};
+
+    use polywrap_core::{uri::Uri, resolution::{uri_resolution_context::{UriResolutionContext, UriPackageOrWrapper}, uri_resolver::UriResolver}, invoker::Invoker, package::GetManifestOptions};
+    use polywrap_msgpack::rmp_serde::encode;
+
+    use super::{UriResolverWrapper, MaybeUriOrManifest};
+
+    struct MockInvoker {}
```

Comment:
can we use the mock invoker from https://github.com/polywrap/rust-client/blob/main/packages/tests-utils/src/mocks/invoker.rs ?

Reply:
i added this :smile: